### ISSUE TITLE
Is a ggml tensor's dimension order reversed?

### ISSUE THREAD

**Issue by chunhualiao** [ggerganov/ggml#710]
Hi, when I look at the model/wte tensor (token embedding weights) of GPT-2 117 M example. 
It has a shape of [768,50257]. I guess it is the [embedding_dim, vocabulary_size]. 

Should the usual dimension order for wte be [vocabulary_size, embedding_dim] instead?
If so, why does ggml tensor store the dimensions in a reversed order?

Similarly,  model/wpe (positional encoding weights) has the shape of [768,1024], which also seems to be reversed from a usual [1024, 768] order. 

Thanks,

**Comment by ggerganov** [comment]
Yes, dimension order is reversed compared to Python: https://github.com/ggerganov/ggml/issues/500#issuecomment-1704322898
